spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java

```diff
@@ -92,11 +91,32 @@ public CloudSqlJdbcInfoProvider appengineCloudSqlJdbcInfoProvider() {
 
 	@Bean
 	@ConditionalOnMissingBean(CloudSqlJdbcInfoProvider.class)
-	public CloudSqlJdbcInfoProvider defaultJdbcInfoProvider() {
-		CloudSqlJdbcInfoProvider defaultProvider = new DefaultCloudSqlJdbcInfoProvider(this.gcpCloudSqlProperties);
+	@ConditionalOnClass(com.google.cloud.sql.mysql.SocketFactory.class)
```

Comment:
`ConditionalOnClass`  on a method is an anti-pattern. It should be added to a `class` (it can be inner` so that the JVM doesn't load it at all if the target class isn't present.

Reply:
Then the only alternative I see is to create two SQL auto configuration classes: `GcpCloudSqlMySqlAutoConfiguration` and `GcpCloudSqlPostgreSqlAutoConfiguration`. I'm not sure I like this approach, there would be a lot of code duplication...

WDYT?

Reply:
Having auto-configuration for MySQL and PostgreSQL in the same class is a smell anyway. I am sure there is a way to restructure the code to avoid the duplication. Nothing Spring Boot related there IMO.